zio-http-datastar-sdk/src/main/scala/zio/http/datastar/ServerSentEventGenerator.scala

```diff
@@ -0,0 +1,195 @@
+package zio.http.datastar
+
+import zio._
+
+import zio.http.ServerSentEvent
+import zio.http.template2._
+
+sealed trait EventType extends Product with Serializable
+
+object EventType {
+  case object PatchElements extends EventType {
+    override def toString: String = "datastar-patch-elements"
+  }
+  case object PatchSignals  extends EventType {
+    override def toString: String = "datastar-patch-signals"
+  }
+}
```

Comment:
I always prefer avoiding using `toString`. You defined a `render` function in the template2 code. Maybe it could be better to reuse this naming

```suggestion
sealed trait EventType extends Product with Serializable {
  def render: String
}

object EventType {
  case object PatchElements extends EventType {
    override def render: String = "datastar-patch-elements"
  }
  case object PatchSignals  extends EventType {
    override def render: String = "datastar-patch-signals"
  }
}
```